compiler/src/iree/compiler/Dialect/Stream/Transforms/test/clone_to_consumers.mlir

```diff
@@ -138,15 +137,34 @@ util.func private @uniformMultiResultDispatchOp() -> (tensor<1xi32>, tensor<1xi3
 // CHECK-LABEL: @multipleUses
 util.func private @multipleUses() -> (tensor<1xi32>, tensor<1xi32>) {
   %splat_value = arith.constant 123 : i32
-  //      CHECK: %[[SPLAT_A:.+]] = flow.tensor.splat
+  //      CHECK: %[[SPLAT_B:.+]] = flow.tensor.splat
+  // CHECK-NEXT: %[[SPLAT_A:.+]] = flow.tensor.splat
   %splat = flow.tensor.splat %splat_value : tensor<1xi32>
   // CHECK-NEXT: %[[DISPATCH_A:.+]] = flow.dispatch @ex::@a(%[[SPLAT_A]], %[[SPLAT_A]])
   %dispatch = flow.dispatch @ex::@a(%splat, %splat) {stream.affinity = #hal.device.promise<@dev_a>} : (tensor<1xi32>, tensor<1xi32>) -> tensor<1xi32>
   // CHECK-NEXT: %[[TRANSFER_A:.+]] = flow.tensor.transfer %[[DISPATCH_A]]
   %transfer_a = flow.tensor.transfer %dispatch : tensor<1xi32> to #hal.device.promise<@dev_a>
-  // CHECK-NEXT: %[[SPLAT_B:.+]] = flow.tensor.splat
   // CHECK-NEXT: %[[TRANSFER_B:.+]] = flow.tensor.transfer %[[SPLAT_B]]
   %transfer_b = flow.tensor.transfer %splat : tensor<1xi32> to #hal.device.promise<@dev_b>
   // CHECK-NEXT: util.return %[[TRANSFER_A]], %[[TRANSFER_B]]
   util.return %transfer_a, %transfer_b : tensor<1xi32>, tensor<1xi32>
 }
+
+// -----
+
+// Tests that splats with only one use are never cloned.
+
+// CHECK:       iterationCount: 0
```

Comment:
Don't rely on debug prints in tests - instead, check what you're actually testing. Here, if you don't expect the splat to be cloned then test that the splat is not present twice.